File: graphrag/config/models/input_config.py
# Copyright (c) 2024 Microsoft Corporation.
# Licensed under the MIT License

"""Parameterization settings for the default configuration."""

from pydantic import BaseModel, Field

import graphrag.config.defaults as defs
from graphrag.config.defaults import graphrag_config_defaults
from graphrag.config.enums import InputFileType
from graphrag.config.models.storage_config import StorageConfig


class InputConfig(BaseModel):
    """The default configuration section for Input."""

    storage: StorageConfig = Field(
        description="The storage configuration to use for reading input documents.",
        default=StorageConfig(
            base_dir=graphrag_config_defaults.input.storage.base_dir,
        ),
    )
    file_type: InputFileType = Field(
        description="The input file type to use.",
        default=graphrag_config_defaults.input.file_type,
    )
    encoding: str = Field(
        description="The input file encoding to use.",
        default=defs.graphrag_config_defaults.input.encoding,
    )
    file_pattern: str = Field(
        description="The input file pattern to use.",
        default=graphrag_config_defaults.input.file_pattern,
    )
    file_filter: dict[str, str] | None = Field(
        description="The optional file filter for the input files.",
        default=graphrag_config_defaults.input.file_filter,
    )
    text_column: str = Field(
        description="The input text column to use.",
        default=graphrag_config_defaults.input.text_column,
    )
    title_column: str | None = Field(
        description="The input title column to use.",
        default=graphrag_config_defaults.input.title_column,
    )
    metadata: list[str] | None = Field(
        description="The document attribute columns to use.",
        default=graphrag_config_defaults.input.metadata,
    )

File: src/ManagedCode.GraphRag/Config/InputConfig.cs
using System.Collections.Generic;

namespace GraphRag.Config;

public sealed class InputConfig
{
    public StorageConfig Storage { get; set; } = new()
    {
        BaseDir = "input",
        Type = StorageType.File
    };

    public InputFileType FileType { get; set; } = InputFileType.Text;

    public string Encoding { get; set; } = "utf-8";

    public string FilePattern { get; set; } = ".*\\.txt$";

    public Dictionary<string, string>? FileFilter { get; set; }
        = new(StringComparer.OrdinalIgnoreCase);

    public string TextColumn { get; set; } = "text";

    public string? TitleColumn { get; set; } = "title";

    public List<string>? Metadata { get; set; }
        = new() { "publication" };
}
